### PR TITLE
Apply plugintools and registry to existing plugins

### PR DESCRIPTION
**Pull request by edwarnicke**

### CODE REVIEW

**Comment by edwarnicke** [comment]
Note: This is built on #190

**Comment by edwarnicke** [comment]
Note: Checks have not passed... Travis was not triggered.

**Comment by edwarnicke** [comment]
OK, checks have really passed now... review at will :)

File: utils/helper/plugintools/close.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	return func() error {
+		err := deptools.Close(p)
+		if err != nil {
+			return err
+		}
+		registry.Shared().Delete(p)
+		for _, close := range closes {
+			err = close()
+			if err != nil {
+				return err
```
Comment (fkautz) [inline utils/helper/plugintools/close.go]: Do we want to stop here or preserve the error and keep trying to close the rest of the range?

File: utils/helper/plugintools/close.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	return func() error {
+		start := time.Now()
+		if log == nil {
+			return fmt.Errorf("argument log to plugintools.LoggingCloseFunc cannot be nil")
+		}
+		err := Close(p, inits...)
+		stop := time.Now()
+		duration := stop.Sub(start)
+		if err != nil {
+			log.Infof("Close Failed after %v: %s", duration, err)
```
Comment (fkautz) [inline utils/helper/plugintools/close.go]: lower case messages
Close => close
Failed => failed

File: utils/helper/plugintools/init.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	return func() error {
+		err := deptools.Check(p)
+		if err != nil {
+			return err
+		}
+		err = deptools.Init(p)
+		if err != nil {
+			return err
+		}
+		for _, init := range inits {
```
Comment (fkautz) [inline utils/helper/plugintools/init.go]: Do we want to close current inited plugins if one fails?

File: utils/helper/plugintools/close.go (hunk 1 above)
Comment (edwarnicke) [reply utils/helper/plugintools/close.go]: Done in latest commit.

File: utils/helper/plugintools/init.go (hunk 3 above)
Comment (edwarnicke) [reply utils/helper/plugintools/init.go]: Done in current commit.

File: utils/helper/plugintools/close.go (hunk 2 above)
Comment (edwarnicke) [reply utils/helper/plugintools/close.go]: Done in latest commit.

**Comment by sbezverk** [comment]
After rebasing on this PR, I am getting this panic on new plugin:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0xa47712]
goroutine 1 [running]:
panic(0x117a0e0, 0x1bffd60)
	/usr/local/go/src/runtime/panic.go:551 +0x3c1 fp=0xc420027978 sp=0xc4200278d8 pc=0x42b071
runtime.panicmem()
	/usr/local/go/src/runtime/panic.go:63 +0x5e fp=0xc420027998 sp=0xc420027978 pc=0x429f4e
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:388 +0x17a fp=0xc4200279e8 sp=0xc420027998 pc=0x44015a
github.com/ligato/networkservicemesh/plugins/logger.(*Plugin).Infof(0xc4200f4fc0, 0x1310970, 0x22, 0xc4202b4220, 0x2, 0x2)
	<autogenerated>:1 +0x32 fp=0xc420027a28 sp=0xc4200279e8 pc=0xa47712
github.com/ligato/networkservicemesh/utils/helper/plugintools.LoggingInitFunc.func1(0xc400000008, 0x1380728)
	/go/src/github.com/ligato/networkservicemesh/utils/helper/plugintools/init.go:66 +0x1fd fp=0xc420027ae0 sp=0xc420027a28 pc=0xa49f3d
github.com/ligato/networkservicemesh/utils/idempotent.(*Impl).IdempotentInit(0xc4201b9930, 0xc4204e2b00, 0x0, 0x0)
	/go/src/github.com/ligato/networkservicemesh/utils/idempotent/idempotent_impl.go:95 +0xb6 fp=0xc420027b10 sp=0xc420027ae0 pc=0x5b9846
github.com/ligato/networkservicemesh/plugins/finalizer.(*Plugin).Init(0xc4201b9930, 0x12637c0, 0xc4201b9930)
	/go/src/github.com/ligato/networkservicemesh/plugins/finalizer/finalizer_plugin.go:80 +0x119 fp=0xc420027b70 sp=0xc420027b10 pc=0xfa8b39
github.com/ligato/networkservicemesh/utils/helper/deptools.initDeps(0xc4203aa060, 0xc42017bae0, 0xc4203e40e0)
	/go/src/github.com/ligato/networkservicemesh/utils/helper/deptools/init.go:49 +0x1a8 fp=0xc420027c28 sp=0xc420027b70 pc=0x9c5118
github.com/ligato/networkservicemesh/utils/helper/deptools.Init(0x1417300, 0xc42017bae0, 0x0, 0x0)
	/go/src/github.com/ligato/networkservicemesh/utils/helper/deptools/init.go:35 +0x78 fp=0xc420027c68 sp=0xc420027c28 pc=0x9c4f18
github.com/ligato/networkservicemesh/utils/helper/plugintools.InitFunc.func1(0x1417300, 0xc42017bae0)
	/go/src/github.com/ligato/networkservicemesh/utils/helper/plugintools/init.go:41 +0x71 fp=0xc420027cc8 sp=0xc420027c68 pc=0xa49c81
github.com/ligato/networkservicemesh/utils/helper/plugintools.Init(0x1417300, 0xc42017bae0, 0xc4203e0000, 0x1, 0x1, 0x30, 0xc420000180)
	/go/src/github.com/ligato/networkservicemesh/utils/helper/plugintools/init.go:31 +0x5d fp=0xc420027d08 sp=0xc420027cc8 pc=0xa4957d
github.com/ligato/networkservicemesh/utils/helper/plugintools.LoggingInitFunc.func1(0xc400000008, 0x1380728)
	/go/src/github.com/ligato/networkservicemesh/utils/helper/plugintools/init.go:62 +0xc9 fp=0xc420027dc0 sp=0xc420027d08 pc=0xa49e09
github.com/ligato/networkservicemesh/utils/idempotent.(*Impl).IdempotentInit(0xc42017bae0, 0xc4203ac040, 0x0, 0x0)
	/go/src/github.com/ligato/networkservicemesh/utils/idempotent/idempotent_impl.go:95 +0xb6 fp=0xc420027df0 sp=0xc420027dc0 pc=0x5b9846
github.com/ligato/networkservicemesh/plugins/nsmcommand.(*Plugin).Init(0xc42017bae0, 0x0, 0xc4203a8060)
	/go/src/github.com/ligato/networkservicemesh/plugins/nsmcommand/impl.go:32 +0xe6 fp=0xc420027e48 sp=0xc420027df0 pc=0x1049a06
github.com/ligato/networkservicemesh/plugins/interupthandler.(*Plugin).init(0xc42033bb80, 0xc42033bb80, 0xc420121ee0)
	/go/src/github.com/ligato/networkservicemesh/plugins/interupthandler/impl.go:45 +0x11c fp=0xc420027eb0 sp=0xc420027e48 pc=0x5ba04c
github.com/ligato/networkservicemesh/plugins/interupthandler.(*Plugin).(github.com/ligato/networkservicemesh/plugins/interupthandler.init)-fm(0xc400000008, 0x1380728)
	/go/src/github.com/ligato/networkservicemesh/plugins/interupthandler/impl.go:34 +0x2a fp=0xc420027ed8 sp=0xc420027eb0 pc=0x5ba31a
github.com/ligato/networkservicemesh/utils/idempotent.(*Impl).IdempotentInit(0xc42033bb80, 0xc420121f28, 0x0, 0x0)
	/go/src/github.com/ligato/networkservicemesh/utils/idempotent/idempotent_impl.go:95 +0xb6 fp=0xc420027f08 sp=0xc420027ed8 pc=0x5b9846
github.com/ligato/networkservicemesh/plugins/interupthandler.(*Plugin).Init(0xc42033bb80, 0xc42033bb80, 0x0)
	/go/src/github.com/ligato/networkservicemesh/plugins/interupthandler/impl.go:34 +0x48 fp=0xc420027f48 sp=0xc420027f08 pc=0x5b9f08
main.main()
	/go/src/github.com/ligato/networkservicemesh/cmd/netmesh/main.go:30 +0xc1 fp=0xc420027f88 sp=0xc420027f48 pc=0x104a891
runtime.main()
	/usr/local/go/src/runtime/proc.go:198 +0x212 fp=0xc420027fe0 sp=0xc420027f88 pc=0x42cd82
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc420027fe8 sp=0xc420027fe0 pc=0x457e61
goroutine 2 [force gc (idle)]:
```

**Pull request merged**